### ISSUE TITLE
2019-7-26

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#67]
181.Do mid variables need declaration? No, pred is an example; (2019-6-20)
182.Use -1 is very convenient; initialize X, Y and theta; try to use tf.zeros, done; (2019-6-20)
183.bad operand type for unary -: 'Dimension'; solved; (2019-6-20)
184.The problem is how to deal with matrix; let us rewrite linear regression as matrix; use another way to write zip; done; (2019-6-20)
185.Can we use matrix to run optimizer? Done; (2019-6-20) 
186.Strange, cost is nan; use tensorflow’s way to write cost; done; why cost still equals to nan? You are overflowing float32 because the learning rate is too high for your problem, and instead of converging the weight variable (W) is oscillating towards larger and larger magnitudes on each step of gradient descent; (2019-6-20)
187.Do we have better optimize method? Give up; (2019-6-20)
188.Let’s see RNN model; search zhihu about RNN; 
189.一文搞懂RNN（循环神经网络）基础篇; 循环神经网络的隐藏层的值s不仅仅取决于当前这次的输入x，还取决于上一次隐藏层的值s。权重矩阵 W就是隐藏层上一次的值作为这一次的输入的权重。
